src/Utilities/TrackedDependencies/CanonicalTrackedInputFiles.cs

```diff
@@ -49,6 +49,8 @@ public class CanonicalTrackedInputFiles
         private readonly HashSet<string> _excludedInputPaths = new HashSet<string>(StringComparer.Ordinal);
         // Cache of last write times
         private readonly ConcurrentDictionary<string, DateTime> _lastWriteTimeCache = new ConcurrentDictionary<string, DateTime>(StringComparer.Ordinal);
+        // Cache of files that have been checked and exist.
+        private HashSet<string> fileCache = new HashSet<string>();
```

Comment:
What's the lifetime of this class? Does it exist only for a single computation (so the lifetime of the whole cache is short, and we don't have to worry about invalidating it) or for longer (and there might be stale cached data)?